From 0cb467d86df393fc2c577fd707a951c5dfa5fdc8 Mon Sep 17 00:00:00 2001
From: Kevin Lampis <kevin.lampis@citrix.com>
Date: Tue, 13 Jan 2026 15:47:17 +0100
Subject: x86: fix incorrect return value for has_if_pschange_mc

The Atom block used to fall through into the Phi block to get to it's
return false, and this was accidentally dropped with the Phi removal.

Fixes: 85191cf32180 ("x86: drop Xeon Phi support")
Signed-off-by: Kevin Lampis <kevin.lampis@citrix.com>
Reviewed-by: Andrew Cooper <andrew.cooper3@citrix.com>
master commit: a2a34d76643e49ccc949296c9a45888034e50b55
master date: 2026-01-12 13:07:36 +0100

diff --git a/xen/arch/x86/hvm/vmx/vmx.c b/xen/arch/x86/hvm/vmx/vmx.c
index 962b7f543c3d..338dd794227b 100644
--- a/xen/arch/x86/hvm/vmx/vmx.c
+++ b/xen/arch/x86/hvm/vmx/vmx.c
@@ -2979,6 +2979,7 @@ static bool __init has_if_pschange_mc(void)
     case 0x75: /* Lightning Mountain */
     case 0x7a: /* Gemini Lake */
     case 0x86: /* Jacobsville */
+        return false;
 
     default:
         printk("Unrecognised CPU model %#x - assuming vulnerable to IF_PSCHANGE_MC\n",
